### PR TITLE
WIP: add redux route

### LINKED ISSUE #20: Complete React Routing
Complete Routing as follows:

- When user enters '/', if the user is logged in, he is redirect to '/app/files'
- When user enters '/', if the user is not logged in, he is redirected to '/auth'

Ensure the user cannot access '/app' without logging in!

EXTREMELY USEFUL LINK: https://css-tricks.com/react-router-4/?fbclid=IwAR0cL1etRcnn_m80hE-_q8ola5MKIMYLHKjMTTzEniBSU3bRRhZY_b1zXLE

### PR DESCRIPTION
**Pull request by virajg224**
**Mostly working. Need a log out to test fully.**

Done
- Redirects from Auth to Main if logged in
- Saves token in localStorage (need to look at other methods)
- stores user and userToken in redux state
- if you access any auth routes, it checks token otherwise redirects to auth page

Need to do:
- clean up code
- look for logical improvements
- redirect any other path to auth
- work on logout after logout button is complete
- verify authToken?

### CODE REVIEW

**Comment by virajg224** [comment]
Resolve #27

File: contrail-ui/src/components/auth/auth-page/Auth.tsx [hunk 1]
```diff
@@ -17,9 +30,10 @@ export default class Auth extends Component<{}, IAuthState> {
 
   public render() {
     let renderForm;
+    console.log(this)
```
Comment (notoriousmango) [inline contrail-ui/src/components/auth/auth-page/Auth.tsx]: delete this

File: contrail-ui/src/components/auth/login-form/LoginForm.tsx [hunk 2]
```diff
@@ -8,28 +8,48 @@ import TextField from "@material-ui/core/TextField";
...
 import { loginUser } from "../../../utils/auth-utils";
 import styles from "../authStyles";
 import { LoginFormProps, LoginFormState } from "./loginForm.type";
 
 class LoginForm extends Component<LoginFormProps, LoginFormState> {
-    public state = {
-        email: "",
-        password: "",
-    };
+    constructor(props: any) {
```
Comment (notoriousmango) [inline contrail-ui/src/components/auth/login-form/LoginForm.tsx]: constructor(props: LoginFormProps)

File: contrail-ui/src/components/auth/login-form/LoginForm.tsx [hunk 3]
```diff
@@ -8,28 +8,48 @@ import TextField from "@material-ui/core/TextField";
...
+        .catch((err) => {
+            // display message
+        });
     }
 
     public render() {
-    const { classes, toggleForm } = this.props;
-
+    const { classes, toggleForm, authToken} = this.props;
+    console.log(this)
```
Comment (notoriousmango) [inline contrail-ui/src/components/auth/login-form/LoginForm.tsx]: delete this

File: contrail-ui/src/components/auth/auth-page/Auth.tsx [hunk 4]
```diff
@@ -1,12 +1,25 @@
...
+import * as ROUTES from "../../../routes";
 import LoginForm from "../login-form/LoginForm";
 import RegisterForm from "../register-form/RegisterForm";
 import { IAuthState } from "./auth.type";
 
-export default class Auth extends Component<{}, IAuthState> {
-  public state = {
-    displayForm: "LoginForm",
-  };
+class Auth extends Component<any, IAuthState> {
```
Comment (1298se) [inline contrail-ui/src/components/auth/auth-page/Auth.tsx]: Use state = instead of constructor since it is newer

File: contrail-ui/tslint.json [hunk 5]
```diff
@@ -5,6 +5,8 @@
         "tslint-react"
     ],
     "jsRules": {},
-    "rules": {},
+    "rules": {
+        "object-literal-sort-keys": false
```
Comment (1298se) [inline contrail-ui/tslint.json]: this is illegal

File: contrail-ui/src/App.tsx [hunk 6]
```diff
@@ -2,12 +2,18 @@ import React from "react";
...
+import * as ROUTES from "./routes";
 
 function App() {
   return (
     <Router>
-      <Route path="/auth" exact={true} component={Auth} />
-      <Route path="/main" exact={true} component={MainFrame} />
+      <Route path={ROUTES.AUTH} component={Auth} />
+      <AuthorizedRoute path={ROUTES.MAIN} component={MainFrame} />
+      <AuthorizedRoute path={ROUTES.FILES} component={MainFrame} authRoute={"files"} />
```
Comment (1298se) [inline contrail-ui/src/App.tsx]: Does this mean for every single authorized route we have to add it here?

File: contrail-ui/src/routes.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export const MAIN = "/main";
+export const AUTH = "/auth";
+export const FILES = "/main/files";
```
Comment (1298se) [inline contrail-ui/src/routes.ts]: do we have to do this for every route?

File: contrail-ui/src/routes.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+export const FAVORITES = "/main/favorites";
+export const SHARED = "/main/shared";
```
Comment (1298se) [inline contrail-ui/src/routes.ts]: change /main to /app

File: contrail-ui/src/App.tsx (hunk 6 above)
Comment (notoriousmango) [reply contrail-ui/src/App.tsx]: isn't there like a children route option in react?

**Comment by virajg224** [comment]
I fixed most of the comments except the routing ones. This method of routing works. I will have to do more research to see if there is a better way to route

File: contrail-ui/src/App.tsx (hunk 6 above)
Comment (virajg224) [reply contrail-ui/src/App.tsx]: Yeah we can edit this later when we have the children components ready. We just move these into those components and only have /auth and /app here

File: contrail-ui/tslint.json (hunk 5 above)
Comment (virajg224) [reply contrail-ui/tslint.json]: Redux standard is that we type field at the top and this rule causes an error for it

File: contrail-ui/src/routes.ts (hunk 7 above)
Comment (virajg224) [reply contrail-ui/src/routes.ts]: this is just constants

**Pull request merged**